storage/src/backend/rocksdb_backend.rs

```diff
@@ -2,36 +2,26 @@
 // SPDX-License-Identifier: MIT
 
 use crate::merkle_storage::{ContextValue, EntryHash};
-use crate::persistent::database::GetInMemStats;
-use crate::persistent::database::KeyValueStoreWithSchema;
-use crate::persistent::database::RocksDBStats;
-use crate::storage_backend::{StorageBackend, StorageBackendError};
+
+use crate::persistent::database::{DBError, KeyValueStoreBackend};
+use crate::storage_backend::NotGarbageCollected;
 use crate::MerkleStorage;
-use rocksdb::WriteBatch;
-use rocksdb::{WriteOptions, DB};
+use rocksdb::DB;
 use serde::{Deserialize, Serialize};
 use std::ops::Deref;
 use std::sync::Arc;
 
 pub struct RocksDBBackend {
-    column_name: &'static str,
     inner: Arc<DB>,
 }
 
 impl RocksDBBackend {
-    pub fn new(db: Arc<DB>, column_name: &'static str) -> Self {
-        RocksDBBackend {
-            inner: db,
-            column_name,
-        }
+    pub fn new(db: Arc<DB>) -> Self {
+        RocksDBBackend { inner: db }
     }
-}
 
-impl RocksDBBackend {
-    fn default_write_options() -> WriteOptions {
-        let mut opts = WriteOptions::default();
-        opts.set_sync(false);
-        opts
+    fn merkle_ref(&self) -> &dyn KeyValueStoreBackend<MerkleStorage> {
+        self.inner.deref() as &dyn KeyValueStoreBackend<MerkleStorage>
```

Comment:
@mat13mn what prevents the creation of a newtype to avoid the use of `DB` directly? this here just hides the problem under the rug instead of fixing the ambiguity. I see a few other places where `as &dyn` casts are still being made.

Reply:
@mat13mn @bkontur although I am commenting here to understand why things are currently like that, this refactoring I'm suggesting here (and on the older comment) is probably not to be made as part of this PR because there is a lot going on already. Just trying to understand if there is anything that prevents us from taking the cleaner approach.

Reply:
@tizoc @mat13mn 
I am not sure about that dyn stuff, and we use it in different place,
last week I create anther jira which is related to this: https://simplestakingcom.atlassian.net/browse/TE-437
we can do it as as part of this jira,

just pls. add to the description, what we should look at